src/main/java/com/epam/aidial/core/controller/ResourceController.java

```diff
@@ -158,10 +167,23 @@ private Future<?> putResource(ResourceDescription descriptor) {
                     switch (resourceType) {
                         case PROMPT -> ProxyUtil.convertToObject(body, Prompt.class);
                         case CONVERSATION -> ProxyUtil.convertToObject(body, Conversation.class);
+                        case APPLICATION -> {
+                            Application application = ProxyUtil.convertToObject(body, Application.class, true);
+                            if (application != null) {
+                                // replace application name with it's url
+                                application.setName(descriptor.getUrl());
+                                // defining user roles in custom applications are not allowed
+                                application.setUserRoles(null);
```

Comment:
It should fail instead of silently resetting the fields.

Reply:
this is not a part of request body, otherwise we need to use another class

Reply:
I prefer to introduce another class with clear structure if you want to